### PR TITLE
Adapt Annotation1DPeakItem to XLMS labels

### PR DESCRIPTION
**Pull request by enetz**
I added some changes that make this work perfectly with my XLMS labels.
Not sure what consequences this might have for other labels.

### CODE REVIEW

**Comment by enetz** [comment]
@timosachsenberg will this work? or should I redo the PR?

**Comment by timosachsenberg** [comment]
hmm as we don't have automated tests for that can you test manually?

**Comment by timosachsenberg** [comment]
@enetz matching the whole string seems a bit dangerous. in this case I would prefer to handle a case for cross-linking data with exact positional matching (e.g., pos == 0)  as done before.

**Comment by enetz** [comment]
@timosachsenberg ok, the match position is constrained to 3 positions now. This now works (mostly) for my data and the RNPxl test files.

**Comment by timosachsenberg** [comment]
hmm seems like a hack... can you include a proper RegEx that tests for your annotations?

**Comment by enetz** [comment]
ok, will try

**Comment by enetz** [comment]
I added a more specific regexp for my XLMS labels.

File: src/openms_gui/source/VISUAL/ANNOTATION/Annotation1DPeakItem.cpp [hunk 1]
```diff
@@ -227,15 +227,29 @@ namespace OpenMS
...
       {
         QString index_str = reg_exp.cap(1);
 
         // put sub html tag around number
-        text = text.left(match_pos)
-                + text[match_pos]
+        text = text[match_pos]
                 + QString("<sub>") + index_str + QString("</sub>")
                 + text.right(text.size() - match_pos - index_str.size() - 1);
+      } else {
```
Comment (timosachsenberg) [inline src/openms_gui/source/VISUAL/ANNOTATION/Annotation1DPeakItem.cpp]: Bracket on new line? Style

File: src/openms_gui/source/VISUAL/ANNOTATION/Annotation1DPeakItem.cpp [hunk 2]
```diff
@@ -226,17 +226,40 @@ namespace OpenMS
...
       }
 
       // charge
+      text.replace(QRegExp("\\+9$"), "<sup>9+</sup>");
+      text.replace(QRegExp("\\+8$"), "<sup>8+</sup>");
+      text.replace(QRegExp("\\+7$"), "<sup>7+</sup>");
+      text.replace(QRegExp("\\+6$"), "<sup>6+</sup>");
+      text.replace(QRegExp("\\+5$"), "<sup>5+</sup>");
+      text.replace(QRegExp("\\+4$"), "<sup>4+</sup>");
+      text.replace(QRegExp("\\+3$"), "<sup>3+</sup>");
```
Comment (timosachsenberg) [inline src/openms_gui/source/VISUAL/ANNOTATION/Annotation1DPeakItem.cpp:264]: Why is this difference again? +n and n+

**Pull request merged**